connector/signaltometrics/config/config.go

```diff
@@ -0,0 +1,337 @@
+		}
+	}
+	return nil
+}
+
+func (mi *MetricInfo) validateSummary() error {
+	if mi.Summary != nil {
+		if mi.Summary.Value == "" {
+			return errors.New("value OTTL statement is required")
+		}
+	}
+	return nil
+}
+
+func (mi *MetricInfo) validateSum() error {
+	if mi.Sum != nil {
+		if mi.Sum.Double != "" && mi.Sum.Int != "" {
+			return errors.New("only one of double or int must be defined for sum metrics")
+		}
+		if mi.Sum.Double == "" && mi.Sum.Int == "" {
+			return errors.New("one of double or int must be defined for sum metrics")
+		}
+	}
+	return nil
+}
+
+func (mi *MetricInfo) ensureDefaults() {
+	if mi.Explicit != nil {
+		// Add default buckets if explicit histogram is defined
+		if len(mi.Explicit.Buckets) == 0 {
+			mi.Explicit.Buckets = defaultHistogramBuckets[:]
+		}
+	}
+	if mi.Exponential != nil {
+		if mi.Exponential.MaxSize == 0 {
+			mi.Exponential.MaxSize = defaultExponentialHistogramMaxSize
+		}
+	}
+}
+
+func validateSpanOTTLStatement(statements []string) error {
+	parser, err := ottlspan.NewParser(
+		ottlget.CustomFuncs[ottlspan.TransformContext](),
+		component.TelemetrySettings{Logger: zap.NewNop()},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create parser for OTTL spans: %w", err)
+	}
+	if _, err := parser.ParseStatements(statements); err != nil {
+		return fmt.Errorf("failed to parse span OTTL statements: %w", err)
+	}
+	return nil
+}
+
+func validateDatapointOTTLStatement(statements []string) error {
+	parser, err := ottldatapoint.NewParser(
+		ottlget.CustomFuncs[ottldatapoint.TransformContext](),
+		component.TelemetrySettings{Logger: zap.NewNop()},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create parser for OTTL datapoints: %w", err)
+	}
+	if _, err := parser.ParseStatements(statements); err != nil {
+		return fmt.Errorf("failed to parse datapoint OTTL statements: %w", err)
+	}
+	return nil
+}
+
+func validateLogOTTLStatement(statements []string) error {
+	parser, err := ottllog.NewParser(
+		ottlget.CustomFuncs[ottllog.TransformContext](),
+		component.TelemetrySettings{Logger: zap.NewNop()},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create parser for OTTL logs: %w", err)
+	}
+	if _, err := parser.ParseStatements(statements); err != nil {
+		return fmt.Errorf("failed to parse log OTTL statements: %w", err)
+	}
+	return nil
+}
+
+type Attribute struct {
+	Key          string `mapstructure:"key"`
+	DefaultValue any    `mapstructure:"default_value"`
+}
+
+type ExplicitHistogram struct {
+	Buckets []float64 `mapstructure:"buckets"`
+	Count   string    `mapstructure:"count"`
+	Value   string    `mapstructure:"value"`
+}
+
+type ExponentialHistogram struct {
+	MaxSize int32  `mapstructure:"max_size"`
+	Count   string `mapstructure:"count"`
+	Value   string `mapstructure:"value"`
+}
+
+type Summary struct {
+	Count string `mapstructure:"count"`
+	Value string `mapstructure:"value"`
+}
+
+type Sum struct {
+	Double string `mapstructure:"double"`
+	Int    string `mapstructure:"int"`
```

Comment:
Why does Sum need an explicit Double/Int distinction, and others don't?

Reply:
Sum can record either integer values or double values, others (histograms and summaries) don't need this since they record double values and integer counts.

Reply:
Could we just have a single `value` configuration and do the right thing depending on what the OTTL returns?

Reply:
Sounds good. If it turns out to be problematic, we could always add another property just for sums like `type: int|double` or something like that.

Reply:
This leads me to another question: What happens if the OTTL expression returns an invalid value, such as a string that can't be coerced to a number? I think what should happen is that we ignore the event. This may happen when using a custom attribute that uses different types across different services.

Reply:
We will ignore but with an error in the logs.

Reply:
Fixed in https://github.com/elastic/opentelemetry-collector-components/pull/169/commits/44a0fe792f5c3a3da8637200cf49461b2856d1df

Reply:
I will create an issue for this.

Reply:
Created an issue for error mode: https://github.com/elastic/opentelemetry-collector-components/issues/173

I will prioritize the follow-up PR for this.